trace/trace.go

```diff
@@ -47,13 +47,7 @@ func StartSpan(c context.Context, operation string, spanOpts ...sentry.SpanOptio
 		spanOpts = append(spanOpts, sentry.WithDescription(functionName))
 	}
 
-	parent := sentry.SpanFromContext(c)
```

Comment:
the parent‘s context does not obtain other context value except `spanContextKey{}`.

Reply:
Yes, other context values like timeout deadline except a chain of spans wasn't carried over when the context has a parent span.
Accordingly, I changed the title for this PR.